themes/default/content/docs/concepts/inputs-outputs/_index.md

```diff
@@ -320,1662 +318,18 @@ In this case, Pulumi is taking the _output_ from one resource and using it as th
-
-{{% /choosable %}}
-{{% choosable language csharp %}}
-
-```csharp
-var cert = new Certificate("cert", new CertificateArgs
-{
-    DomainName = "example",
-    ValidationMethod = "DNS",
-});
-
-var record = new Record("validation", new RecordArgs
-{
-    Records = {
-        cert.DomainValidationOptions.Apply(opts => opts[0].ResourceRecordValue!)
-    },
-    ...
-});
-```
-
-{{% /choosable %}}
-{{% choosable language java %}}
-
-```java
-var cert = new Certificate("cert",
-    CertificateArgs.builder()
-        .domainName("example")
-        .validationMethod("DNS")
-        .build());
-
-var record = new Record("validation",
-    RecordArgs.builder()
-        .records(
-            cert.domainValidationOptions()
-            .applyValue(opts -> opts.get(0).resourceRecordValue().get())
-            .applyValue(String::valueOf)
-            .applyValue(List::of))
-        .build());
-```
-
-{{% /choosable %}}
-
-{{< /chooser >}}
-
-Instead, to make it easier to access simple property and array elements, an {{< pulumi-output >}} lifts the properties of the underlying value, behaving very much like an instance of it. Lift allows you to access properties and elements directly from the {{< pulumi-output >}} itself without needing {{< pulumi-apply >}}. If we return to the above example, we can now simplify it:
+All resources created by Pulumi will have properties which are returned from the cloud provider API. These values are called *outputs*.
 
-{{< chooser language "javascript,typescript,python,go,csharp,java,yaml" >}}
-
-{{% choosable language javascript %}}
-
-```javascript
-let certCertificate = new aws.acm.Certificate("cert", {
-    domainName: "example.com",
-    validationMethod: "DNS",
-});
-let certValidation = new aws.route53.Record("cert_validation", {
-    records: [
-        certCertificate.domainValidationOptions[0].resourceRecordValue
-    ],
-...
-```
+Outputs are values of type {{< pulumi-output >}}, which behave very much like [promises](https://en.wikipedia.org/wiki/Futures_and_promises) or [monads](https://en.wikipedia.org/wiki/Monad_(functional_programming)). Simply put, outputs are a way of representing values that are not initially known but will become available once the infrastructure resource has completed provisioning, which happens *asynchronously*.
 
-{{% /choosable %}}
-{{% choosable language typescript %}}
+Because outputs are asynchronous, their actual plain values are not immediately available. If you need to access an output’s plain value, you can do so using one of the following options:
 
-```typescript
-let certCertificate = new aws.acm.Certificate("cert", {
-    domainName: "example.com",
-    validationMethod: "DNS",
-});
-let certValidation = new aws.route53.Record("cert_validation", {
-    records: [
-        certCertificate.domainValidationOptions[0].resourceRecordValue
-    ],
-...
-```
+- [Apply](/docs/concepts/inputs-outputs/apply/): a callback that receives the plain value and computes a new output
```

Comment:
Should we tweak the wording here to create a tighter alignment/integration with the sidebar headings? For example, "Apply: a callback that receives the plain value and computes a new output, generally used for accessing single values" (or similar). Not completely convinced this is necessary, but wanted to submit it for consideration.